python/lsst/obs/comCam/ingest.py

```diff
@@ -34,23 +50,57 @@ def getInfo(self, filename):
         if runId != phuInfo['run']:
             raise RuntimeError("Expected runId %s, found %s from path %s" % phuInfo['run'], runId, pathname)
 
-        phuInfo['raftId'] = raftId
-        phuInfo['field'] = acquisitionType
-        phuInfo['jobId'] = int(jobId)
-        phuInfo['raft'] = 'R00'
-        phuInfo['ccd'] = sensorLocationInRaft
+        phuInfo['raftId'] = raftId # also in the header - RAFTNAME
+        phuInfo['field'] = acquisitionType # NOT in the header
+        phuInfo['jobId'] = int(jobId) #  NOT in the header, but do we need it?
+        phuInfo['raft'] = 'R00' # do we need this?
+        phuInfo['ccd'] = sensorLocationInRaft # NOT in the header
 
         return phuInfo, infoList
 
-    # Add entry to config.parse.translators in config/ingest.py if needed
-    def translate_ccd(self, md):
-        return md.get("sensorId")       # ... except that isn't actually present
+    def translate_wavelength(self, md):
+        """Translate wavelength provided by teststand readout.
+
+        The teststand driving script asks for a wavelength, and then reads the value back to ensure that
+        the correct position was moved to. This number is therefore read back with sub-nm precision.
+        Typically the position is within 0.005nm of the desired position, so we warn if it's not very
+        close to an integer value.
+
+        Future users should be aware that the HIERARCH MONOCH-WAVELENG key is NOT the requested value, and
+        therefore cannot be used as a cross-check that the wavelength was close to the one requested.
+        The only record of the wavelength that was set is in the original filename.
+
+        Parameters
+        ----------
+        md : `lsst.daf.base.PropertyList or PropertySet`
+            image metadata
+
+        Returns
+        -------
+        wavelength : `int`
+            The recorded wavelength as an int
+        """
+        raw_wl = md.get("MONOWL")
+        wl = int(round(raw_wl))
+        if abs(raw_wl-wl)>=0.1:
+            logger = lsstLog.Log.getLogger('obs.comCam.ingest')
+            logger.warn('Translated signigicatly non-integer wavelength %s', raw_wl)
```

Comment:
I'm guessing your "significantly" comment is an objection to the lack of precision, given that they're ~always going to be non-integer to some extent. I've changed that to a more quantitative statement, using the 0.1nm constant that I'm testing against. Let me know if that's not what you wanted to imply.

For the warning, I think that's the right thing to do. The value is not _necessarily_ wrong, and if downstream it turns into garbage then you want to just see a warning and _what_ goes wrong, rather than it raising at the time.